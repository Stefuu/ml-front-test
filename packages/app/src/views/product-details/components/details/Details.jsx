import React from 'react'
import styled from 'styled-components'
import { theme } from 'styled-tools'
import Flex from '../../../../components/elements/Flex'
import Box from '../../../../components/elements/Box'
import { rem } from 'polished'
import Button from '../../../../components/inputs/Button'
import { formatCurrency, formatDecimals } from '../../../../helpers'

const DescriptionContainer = styled(Box)`
  line-height: 1.3;
  margin-top: ${rem(32)};
  font-Size: ${rem(16)};
  color: ${theme('colors.b40')};
  span {
    white-space: pre-line;
    word-break: break-word;
  }
`

const Price = styled(Box)`
  margin-top: 20px;
  span {
    line-height: 1;
    font-size: ${rem(46)}
  }
  small {
    line-height: 1.6;
    font-size: ${rem(18)}
    vertical-align: top;
    margin-left: 5px;
  }
`

const Details = ({ item }) => (
  <Flex
    flexDirection={['column', 'row']}>
    <Flex
      mr={rem(32)}
      maxWidth={[590, 560, 661]}
      flexDirection='column'>
      <Box height={[200, 400, 600]}>
        <img height='100%' src={item.picture} alt='Product' />
      </Box>
      <Box
        mt={35}
        fontSize={rem(28)}>
        <span>Descripc√≠on del producto</span>
      </Box>
      <DescriptionContainer>
        <span>{item.description}</span>
      </DescriptionContainer>
    </Flex>
    <Flex
      lineHeight={1.3}
      maxWidth={250}
      flexDirection='column'>
      <Box>
        <span>
          {item.condition === 'new' ? 'Nuevo - ' : ''}{item.sold_quantity} vendidos
        </span>
      </Box>
      <Box
        fontWeight='bold'
        fontSize={24}
        mt={5}>
        <span>{item.title}</span>
      </Box>
      <Price>
        <span>$ {formatCurrency(item.price.amount)}</span><small>{formatDecimals(item.price.decimals)}</small>
      </Price>
      <Button
        data-testid='button-buy'
        mt={rem(54)}
        fontSize={rem(18)}
        width={rem(248)}
        height={rem(45)}>
            Comprar
      </Button>
    </Flex>
  </Flex>
)

export default Details
