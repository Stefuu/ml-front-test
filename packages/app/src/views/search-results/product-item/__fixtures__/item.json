{
	"id": "MLA688402084",
	"title": "Notebook Gadnic 15,6 Intel Windows 10 4gb Cloudbook  + Funda",
	"price": {
		"currency": "ARS",
		"amount": 9999,
		"decimals": 0
	},
	"picture": "http://mla-s1-p.mlstatic.com/938057-MLA29221075330_012019-I.jpg",
	"condition": "new",
	"free_shipping": true,
	"sold_quantity": 250,
	"description": "Cod. : NOT000A2\n\n¡COMPRALO AHORA Y RETIRALO HOY MISMO POR NUESTRA SUCURSAL!\n(Sucursal en Microcentro abierta al público en General)\n\nDespachamos tu producto EN EL DÍA a todo el país\nMOTO: Entregamos dentro de las 24hs hábiles.\n\n////////////////////////////////////////////////////////////////\n\nSomos BIDCOM\n\nTienda Oficial GADNIC\n\nMás de 10 años vendiendo en MercadoLibre. Más de 200000 compradores nos eligieron y recomendaron. Fuimos elegidos por MercadoLibre como mejores vendedores de todo el sitio.\n\n////////////////////////////////////////////////////////////////\n\nNotebook Gadnic Lumina 15.6”\n\nACERCA DE ÉSTE PRODUCTO:\n\nLa notebook Gadnic NOT000A2 con Procesador Intel ATOM X528350 (1.44GHZ), Memoria Ram de 4 GB, Capacidad 32GB, Windows 10 Home Edition y Pantalla de 15,6 pulgadas Resolucion Full HD es ideal para acompañarte con tus tareas de todos los días. \nCuenta con Bluetooth 4.0, Cámara frontal y Wifi Integrado para que puedas conectarte desde dónde estés. Además su diseño compacto y peso súper liviando la hacen la notebook perfecta para llevar a todas partes.\n\n////////////////////////////////////////////////////////////////\n\nCARACTERÍSTICAS:\n\nProcesador Intel ATOM X528350 (1.44GHZ)\r\nMemoria Ram: 4 GB\r\nCapacidad 32GB\r\nWindows 10 Home Edition\r\nPantalla de 15,6”\r\nResolucion Full HD: 1920 x 1080 \r\nBluetooth 4.0\r\nCámara frontal\r\nWifi Integrado.\r\nPuertos: 2 Usb, 1 Mini HDMI, 1 Auriculares , 1 Slot de micro SD\r\n\r\nACCESORIOS\r\nCargador\r\nFunda Protectora\r\nMouse\r\n\r\nPUERTOS\r\n2 Usb\r\n1 Mini HDMI\r\n1 Auriculares\r\n1 Slot de memoria micro SD\n\nSE ENTREGA CON:\n\nNotebook Gadnic Lumina\r\nFunda Protectora\r\nMouse Gadnic - Blanco\r\nCargador 220V\r\nPackaging Original GADNIC \r\n\n////////////////////////////////////////////////////////////////\n\nVIDEO DEMOSTRATIVO:\n\nTe invitamos a ver el video que muestra el producto al final de las imágenes superiores de esta publicación.\n\n////////////////////////////////////////////////////////////////\n\nGARANTÍA: Siempre con nosotros. Gestionamos de manera directa las garantías de todos los productos que comercializamos.\n\n1 AÑO DE GARANTÍA DIRECTA CON NOSOTROS.\n\nServicio Técnico Oficial: Somos Service Oficial de todas las marcas que representamos. \nSi vencida la garantía por alguna razón necesitás servicio técnico, nosotros nos ocupamos. Ante imprevistos te aseguramos que tu producto podrá ser reparado o reemplazado.\n\n////////////////////////////////////////////////////////////////\n\n10 DÍAS DE PRUEBA:\n\nRecibí el producto que esperabas o te devolvemos tu dinero. Estamos tan seguros de la excelencia de nuestros productos que te damos 10 días para que lo pruebes. Si no estás conforme podés devolvernos el producto y te devolveremos el dinero.\n\n////////////////////////////////////////////////////////////////\n\nVENTAS POR MAYOR\n\nEn Bidcom somos Representantes e Importadores directos Gadnic. \n\nConsulte por ventas mayoristas, conviértase en distribuidor de nuestros productos.\n\n////////////////////////////////////////////////////////////////\n\nENVÍOS\n\n* Mercado Envíos\n* Moto en el día: CABA envío GRATIS!; GBA consultar costos\n* OCA a todo el país: GRATIS!\n* Micro a terminal\n\n////////////////////////////////////////////////////////////////\n\nRETIROS EN NUESTRO SHOWROOM:\n\nSucursal Bidcom MICROCENTRO (Cerca de Lavalle y Florida). \nHorarios: Lunes a Viernes de 9 a 19hs / Sábados y Feriados 10 a 17hs\n\nEstamos abiertos al público en general, no es necesario que nos avises antes de venir. Estamos abiertos de corrido en los horarios de atención. Vení a conocer nuestros mejores productos. \n\n////////////////////////////////////////////////////////////////\n\nMEDIOS DE PAGO:\n\nPodés abonar con todos los medios de pago habilitados con Mercado Pago:\n * Tarjeta de Crédito\n * Tarjeta de Débito\n * Pago Fácil\n * Rapi Pago\n * Provincia Net\n * Cobro Express\n * LINK\n * Carga Virtual\n\n¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯\nEmitimos Factura A y B.\nEn el caso de necesitar Factura A por favor comunicalo por el chat interno de Mercado Libre una vez realizada la compra.\n\n////////////////////////////////////////////////////////////////\n\nNUESTROS PRODUCTOS:\n\nEn el siguiente link podés ver todos los productos que desde Bidcom publicamos en MercadoLibre:\n\nhttps://listado.mercadolibre.com.ar/gadnic_Tienda_gadnic_af_to\n\nTablets - Notebooks - Drones - Cámaras de Seguridad - Proyectores - Handies - Cámaras de deportes - TV Box - Contadoras de Billetes - Aspiradoras Robot - Smartwatches - Lentes VR - Patinetas Eléctricas - DJI - Órganos Eléctricos - Autos a Radio Control - Tocadiscos - y mucho más.\n\n////////////////////////////////////////////////////////////////\n\nATENCIÓN: \n¡ÚLTIMAS UNIDADES A PRECIO DE SUPER OFERTA! \n\nComprá ahora y asegurate tu unidad a este precio. ¡Aun Tenemos Stock!\n¡Si podés ver esta publicación es porque TENEMOS STOCK, no hace falta que nos consultes!\n\n¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯\n\nHacenos todas las consultas que tengas, para nosotros es un placer ayudarte.\n\nEstamos Online para todo lo que necesites\n\nBIDCOM\n"
}