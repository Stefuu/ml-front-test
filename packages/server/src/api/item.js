import axios from 'axios'
import logger from '../config/logger'
import { formatItem } from '../components/modules/helpers'
import { baseUrl } from '../components/modules/constants'

export default async (req, res) => {
  const { id } = req.params

  try {
    const itemJson = await axios(`${baseUrl}/items/${id}`)
    const descriptionJson = await axios(`${baseUrl}/items/${id}/description`)
    const data = formatItem(itemJson.data, descriptionJson.data)
    res.setHeader('Content-Type', 'application/json')
    res.json(data)
  } catch (err) {
    logger.error(err)
    if (err && err.response && err.response.status) {
      res.sendStatus(err.response.status)
    } else {
      res.sendStatus(500)
    }
  }
}
