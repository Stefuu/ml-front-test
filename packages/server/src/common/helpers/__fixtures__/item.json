{
    "id": "MLA730963111",
    "site_id": "MLA",
    "title": "Notebook Lenovo Ideapad 330 Dual Core 4gb 500g 14 W10 Mexx",
    "subtitle": null,
    "seller_id": 82916233,
    "category_id": "MLA10261",
    "official_store_id": null,
    "price": 13749,
    "base_price": 13749,
    "original_price": null,
    "currency_id": "ARS",
    "initial_quantity": 11000,
    "available_quantity": 5000,
    "sold_quantity": 25,
    "sale_terms": [],
    "buying_mode": "buy_it_now",
    "listing_type_id": "gold_special",
    "start_time": "2018-06-12T01:02:52.000Z",
    "stop_time": "2038-06-08T12:47:48.000Z",
    "condition": "new",
    "permalink": "https://articulo.mercadolibre.com.ar/MLA-730963111-notebook-lenovo-ideapad-330-dual-core-4gb-500g-14-w10-mexx-_JM",
    "thumbnail": "http://mla-s2-p.mlstatic.com/899873-MLA27546306071_062018-I.jpg",
    "secure_thumbnail": "https://mla-s2-p.mlstatic.com/899873-MLA27546306071_062018-I.jpg",
    "pictures": [
        {
            "id": "899873-MLA27546306071_062018",
            "url": "http://mla-s2-p.mlstatic.com/899873-MLA27546306071_062018-O.jpg",
            "secure_url": "https://mla-s2-p.mlstatic.com/899873-MLA27546306071_062018-O.jpg",
            "size": "500x500",
            "max_size": "1000x1000",
            "quality": ""
        },
        {
            "id": "858575-MLA27546302127_062018",
            "url": "http://mla-s2-p.mlstatic.com/858575-MLA27546302127_062018-O.jpg",
            "secure_url": "https://mla-s2-p.mlstatic.com/858575-MLA27546302127_062018-O.jpg",
            "size": "500x500",
            "max_size": "1000x1000",
            "quality": ""
        },
        {
            "id": "636022-MLA27546270949_062018",
            "url": "http://mla-s2-p.mlstatic.com/636022-MLA27546270949_062018-O.jpg",
            "secure_url": "https://mla-s2-p.mlstatic.com/636022-MLA27546270949_062018-O.jpg",
            "size": "500x500",
            "max_size": "1000x1000",
            "quality": ""
        },
        {
            "id": "921454-MLA27546299657_062018",
            "url": "http://mla-s2-p.mlstatic.com/921454-MLA27546299657_062018-O.jpg",
            "secure_url": "https://mla-s2-p.mlstatic.com/921454-MLA27546299657_062018-O.jpg",
            "size": "500x500",
            "max_size": "1000x1000",
            "quality": ""
        },
        {
            "id": "751342-MLA27546297166_062018",
            "url": "http://mla-s2-p.mlstatic.com/751342-MLA27546297166_062018-O.jpg",
            "secure_url": "https://mla-s2-p.mlstatic.com/751342-MLA27546297166_062018-O.jpg",
            "size": "500x500",
            "max_size": "1000x1000",
            "quality": ""
        },
        {
            "id": "617334-MLA27546297167_062018",
            "url": "http://mla-s2-p.mlstatic.com/617334-MLA27546297167_062018-O.jpg",
            "secure_url": "https://mla-s2-p.mlstatic.com/617334-MLA27546297167_062018-O.jpg",
            "size": "500x500",
            "max_size": "1000x1000",
            "quality": ""
        },
        {
            "id": "988814-MLA27546297169_062018",
            "url": "http://mla-s2-p.mlstatic.com/988814-MLA27546297169_062018-O.jpg",
            "secure_url": "https://mla-s2-p.mlstatic.com/988814-MLA27546297169_062018-O.jpg",
            "size": "500x500",
            "max_size": "1000x1000",
            "quality": ""
        },
        {
            "id": "945475-MLA27546310039_062018",
            "url": "http://mla-s2-p.mlstatic.com/945475-MLA27546310039_062018-O.jpg",
            "secure_url": "https://mla-s2-p.mlstatic.com/945475-MLA27546310039_062018-O.jpg",
            "size": "500x500",
            "max_size": "1000x1000",
            "quality": ""
        }
    ],
    "video_id": null,
    "descriptions": [
        {
            "id": "MLA730963111-1708924178"
        }
    ],
    "accepts_mercadopago": true,
    "non_mercado_pago_payment_methods": [],
    "shipping": {
        "mode": "me2",
        "free_methods": [
            {
                "id": 73328,
                "rule": {
                    "default": true,
                    "free_mode": "country",
                    "free_shipping_flag": true,
                    "value": null
                }
            }
        ],
        "tags": [
            "self_service_in",
            "mandatory_free_shipping"
        ],
        "dimensions": null,
        "local_pick_up": true,
        "free_shipping": true,
        "logistic_type": "cross_docking",
        "store_pick_up": false
    },
    "international_delivery_mode": "none",
    "seller_address": {
        "comment": "",
        "address_line": "",
        "zip_code": "",
        "city": {
            "id": "",
            "name": "Caballito"
        },
        "state": {
            "id": "AR-C",
            "name": "Capital Federal"
        },
        "country": {
            "id": "AR",
            "name": "Argentina"
        },
        "search_location": {
            "neighborhood": {
                "id": "TUxBQkNBQjM4MDda",
                "name": "Caballito"
            },
            "city": {
                "id": "TUxBQ0NBUGZlZG1sYQ",
                "name": "Capital Federal"
            },
            "state": {
                "id": "TUxBUENBUGw3M2E1",
                "name": "Capital Federal"
            }
        },
        "latitude": -34.62237,
        "longitude": -58.450333,
        "id": 102685353
    },
    "seller_contact": null,
    "location": {},
    "geolocation": {
        "latitude": -34.62237,
        "longitude": -58.450333
    },
    "coverage_areas": [],
    "attributes": [
        {
            "id": "BATTERY_TYPE",
            "name": "Tipo de batería",
            "value_id": "94038",
            "value_name": "Ion de litio",
            "value_struct": null,
            "attribute_group_id": "OTHERS",
            "attribute_group_name": "Otros"
        },
        {
            "id": "BRAND",
            "name": "Marca",
            "value_id": "7494",
            "value_name": "Lenovo",
            "value_struct": null,
            "attribute_group_id": "OTHERS",
            "attribute_group_name": "Otros"
        },
        {
            "id": "DIMENSIONS",
            "name": "Medidas",
            "value_id": null,
            "value_name": "0.338 m x 0.25 m x 0.023 m",
            "value_struct": null,
            "attribute_group_id": "OTHERS",
            "attribute_group_name": "Otros"
        },
        {
            "id": "DISPLAY_RESOLUTION",
            "name": "Resolución de la pantalla",
            "value_id": null,
            "value_name": "1366 x 768",
            "value_struct": null,
            "attribute_group_id": "OTHERS",
            "attribute_group_name": "Otros"
        },
        {
            "id": "GPU",
            "name": "GPU",
            "value_id": "-1",
            "value_name": null,
            "value_struct": null,
            "attribute_group_id": "OTHERS",
            "attribute_group_name": "Otros"
        },
        {
            "id": "IS_ULTRABOOK",
            "name": "Es ultrabook",
            "value_id": "242084",
            "value_name": "No",
            "value_struct": null,
            "attribute_group_id": "OTHERS",
            "attribute_group_name": "Otros"
        },
        {
            "id": "ITEM_CONDITION",
            "name": "Condición del ítem",
            "value_id": "2230284",
            "value_name": "Nuevo",
            "value_struct": null,
            "attribute_group_id": "OTHERS",
            "attribute_group_name": "Otros"
        },
        {
            "id": "LINE",
            "name": "Línea",
            "value_id": "27211",
            "value_name": "IdeaPad",
            "value_struct": null,
            "attribute_group_id": "OTHERS",
            "attribute_group_name": "Otros"
        },
        {
            "id": "MODEL",
            "name": "Modelo",
            "value_id": null,
            "value_name": "ideapad 330-14IGM",
            "value_struct": null,
            "attribute_group_id": "OTHERS",
            "attribute_group_name": "Otros"
        },
        {
            "id": "NOTEBOOK_CORES_NUMBER",
            "name": "Cantidad de núcleos de la notebook",
            "value_id": null,
            "value_name": "Dual core",
            "value_struct": null,
            "attribute_group_id": "OTHERS",
            "attribute_group_name": "Otros"
        },
        {
            "id": "OPTICAL_DRIVES",
            "name": "Unidades ópticas",
            "value_id": "6134779",
            "value_name": "No",
            "value_struct": null,
            "attribute_group_id": "OTHERS",
            "attribute_group_name": "Otros"
        },
        {
            "id": "OS_EDITION",
            "name": "Edición del sistema operativo",
            "value_id": "6127363",
            "value_name": "Home",
            "value_struct": null,
            "attribute_group_id": "OTHERS",
            "attribute_group_name": "Otros"
        },
        {
            "id": "OS_NAME",
            "name": "Nombre del sistema operativo",
            "value_id": null,
            "value_name": "Windows 10",
            "value_struct": null,
            "attribute_group_id": "OTHERS",
            "attribute_group_name": "Otros"
        },
        {
            "id": "OS_VERSION",
            "name": "Versión del sistema operativo",
            "value_id": null,
            "value_name": "Home",
            "value_struct": null,
            "attribute_group_id": "OTHERS",
            "attribute_group_name": "Otros"
        },
        {
            "id": "PROCESSOR",
            "name": "Procesador",
            "value_id": "98936",
            "value_name": "Intel Celeron",
            "value_struct": null,
            "attribute_group_id": "OTHERS",
            "attribute_group_name": "Otros"
        },
        {
            "id": "RAM_MEMORY",
            "name": "Memoria RAM",
            "value_id": null,
            "value_name": "4 GB",
            "value_struct": {
                "number": 4,
                "unit": "GB"
            },
            "attribute_group_id": "OTHERS",
            "attribute_group_name": "Otros"
        },
        {
            "id": "RAM_MEMORY_TYPE",
            "name": "Tipo de memoria RAM",
            "value_id": "83822",
            "value_name": "DDR4",
            "value_struct": null,
            "attribute_group_id": "OTHERS",
            "attribute_group_name": "Otros"
        },
        {
            "id": "SCREEN_SIZE",
            "name": "Tamaño de la pantalla",
            "value_id": null,
            "value_name": "14 \"",
            "value_struct": {
                "number": 14,
                "unit": "\""
            },
            "attribute_group_id": "OTHERS",
            "attribute_group_name": "Otros"
        },
        {
            "id": "STORAGE",
            "name": "Capacidad de almacenamiento",
            "value_id": null,
            "value_name": "500 GB",
            "value_struct": {
                "number": 500,
                "unit": "GB"
            },
            "attribute_group_id": "OTHERS",
            "attribute_group_name": "Otros"
        },
        {
            "id": "TYPE_OF_DRIVE",
            "name": "Tipo de almacenamiento de datos",
            "value_id": "365877",
            "value_name": "SSD",
            "value_struct": null,
            "attribute_group_id": "OTHERS",
            "attribute_group_name": "Otros"
        },
        {
            "id": "VIDEO_MEMORY",
            "name": "Memoria de video",
            "value_id": null,
            "value_name": "1 GB",
            "value_struct": {
                "number": 1,
                "unit": "GB"
            },
            "attribute_group_id": "OTHERS",
            "attribute_group_name": "Otros"
        },
        {
            "id": "VIDEO_MEMORY_TYPE",
            "name": "Tipo de memoria de video",
            "value_id": "-1",
            "value_name": null,
            "value_struct": null,
            "attribute_group_id": "OTHERS",
            "attribute_group_name": "Otros"
        },
        {
            "id": "WEBCAM",
            "name": "Cámara web",
            "value_id": "242085",
            "value_name": "Sí",
            "value_struct": null,
            "attribute_group_id": "OTHERS",
            "attribute_group_name": "Otros"
        },
        {
            "id": "WEIGHT",
            "name": "Peso",
            "value_id": null,
            "value_name": "2.1 kg",
            "value_struct": {
                "number": 2.1,
                "unit": "kg"
            },
            "attribute_group_id": "OTHERS",
            "attribute_group_name": "Otros"
        },
        {
            "id": "WITH_TOUCH_SCREEN",
            "name": "Con pantalla táctil",
            "value_id": "242084",
            "value_name": "No",
            "value_struct": null,
            "attribute_group_id": "OTHERS",
            "attribute_group_name": "Otros"
        },
        {
            "id": "WITH_USB",
            "name": "Con USB",
            "value_id": "242085",
            "value_name": "Sí",
            "value_struct": null,
            "attribute_group_id": "OTHERS",
            "attribute_group_name": "Otros"
        }
    ],
    "warnings": [],
    "listing_source": "",
    "variations": [
        {
            "id": 32053903972,
            "price": 13749,
            "attribute_combinations": [
                {
                    "id": "COLOR",
                    "name": "Color",
                    "value_id": "52053",
                    "value_name": "Plateado",
                    "value_struct": null
                }
            ],
            "available_quantity": 5000,
            "sold_quantity": 25,
            "sale_terms": [],
            "picture_ids": [
                "899873-MLA27546306071_062018",
                "858575-MLA27546302127_062018",
                "636022-MLA27546270949_062018",
                "921454-MLA27546299657_062018",
                "751342-MLA27546297166_062018",
                "617334-MLA27546297167_062018",
                "988814-MLA27546297169_062018",
                "945475-MLA27546310039_062018"
            ],
            "seller_custom_field": "81D0000RAR",
            "catalog_product_id": "MLA10389803"
        }
    ],
    "status": "active",
    "sub_status": [],
    "tags": [
        "brand_verified",
        "good_quality_picture",
        "good_quality_thumbnail",
        "immediate_payment",
        "cart_eligible"
    ],
    "warranty": "Garantia Oficial Lenovo Argentina 1 Año.",
    "catalog_product_id": "MLA10636668",
    "domain_id": "MLA-NOTEBOOKS",
    "parent_item_id": null,
    "differential_pricing": null,
    "deal_ids": [],
    "automatic_relist": false,
    "date_created": "2018-06-12T01:02:52.000Z",
    "last_updated": "2019-02-01T15:57:14.618Z",
    "health": 0.9
}