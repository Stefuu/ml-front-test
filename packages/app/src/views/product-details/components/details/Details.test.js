import React from 'react'
import { render } from 'react-testing-library'
import Details from './Details'
import item from './__fixtures__/item.json'

describe('Details', () => {
  it('Should render [snapshot]', async () => {
    const { container } = render(<Details item={item} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
