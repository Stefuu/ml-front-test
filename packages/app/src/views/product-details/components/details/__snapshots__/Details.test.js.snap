// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Should render [snapshot] 1`] = `
<div
  class="sc-bdVaJa ceYYLS"
>
  <div
    class="sc-bdVaJa bHBkUI"
  >
    <div
      class="sc-bdVaJa kWCdxB"
    >
      <img
        alt="Product"
        height="100%"
        src="http://mla-s1-p.mlstatic.com/938057-MLA29221075330_012019-I.jpg"
      />
    </div>
    <div
      class="sc-bdVaJa kuelCM"
    >
      <span>
        Descripcíon del producto
      </span>
    </div>
    <div
      class="sc-bdVaJa sc-htpNat fLbKlL"
    >
      <span>
        Cod. : NOT000A2

¡COMPRALO AHORA Y RETIRALO HOY MISMO POR NUESTRA SUCURSAL!
(Sucursal en Microcentro abierta al público en General)

Despachamos tu producto EN EL DÍA a todo el país
MOTO: Entregamos dentro de las 24hs hábiles.

////////////////////////////////////////////////////////////////

Somos BIDCOM

Tienda Oficial GADNIC

Más de 10 años vendiendo en MercadoLibre. Más de 200000 compradores nos eligieron y recomendaron. Fuimos elegidos por MercadoLibre como mejores vendedores de todo el sitio.

////////////////////////////////////////////////////////////////

Notebook Gadnic Lumina 15.6”

ACERCA DE ÉSTE PRODUCTO:

La notebook Gadnic NOT000A2 con Procesador Intel ATOM X528350 (1.44GHZ), Memoria Ram de 4 GB, Capacidad 32GB, Windows 10 Home Edition y Pantalla de 15,6 pulgadas Resolucion Full HD es ideal para acompañarte con tus tareas de todos los días. 
Cuenta con Bluetooth 4.0, Cámara frontal y Wifi Integrado para que puedas conectarte desde dónde estés. Además su diseño compacto y peso súper liviando la hacen la notebook perfecta para llevar a todas partes.

////////////////////////////////////////////////////////////////

CARACTERÍSTICAS:

Procesador Intel ATOM X528350 (1.44GHZ)
Memoria Ram: 4 GB
Capacidad 32GB
Windows 10 Home Edition
Pantalla de 15,6”
Resolucion Full HD: 1920 x 1080 
Bluetooth 4.0
Cámara frontal
Wifi Integrado.
Puertos: 2 Usb, 1 Mini HDMI, 1 Auriculares , 1 Slot de micro SD

ACCESORIOS
Cargador
Funda Protectora
Mouse

PUERTOS
2 Usb
1 Mini HDMI
1 Auriculares
1 Slot de memoria micro SD

SE ENTREGA CON:

Notebook Gadnic Lumina
Funda Protectora
Mouse Gadnic - Blanco
Cargador 220V
Packaging Original GADNIC 

////////////////////////////////////////////////////////////////

VIDEO DEMOSTRATIVO:

Te invitamos a ver el video que muestra el producto al final de las imágenes superiores de esta publicación.

////////////////////////////////////////////////////////////////

GARANTÍA: Siempre con nosotros. Gestionamos de manera directa las garantías de todos los productos que comercializamos.

1 AÑO DE GARANTÍA DIRECTA CON NOSOTROS.

Servicio Técnico Oficial: Somos Service Oficial de todas las marcas que representamos. 
Si vencida la garantía por alguna razón necesitás servicio técnico, nosotros nos ocupamos. Ante imprevistos te aseguramos que tu producto podrá ser reparado o reemplazado.

////////////////////////////////////////////////////////////////

10 DÍAS DE PRUEBA:

Recibí el producto que esperabas o te devolvemos tu dinero. Estamos tan seguros de la excelencia de nuestros productos que te damos 10 días para que lo pruebes. Si no estás conforme podés devolvernos el producto y te devolveremos el dinero.

////////////////////////////////////////////////////////////////

VENTAS POR MAYOR

En Bidcom somos Representantes e Importadores directos Gadnic. 

Consulte por ventas mayoristas, conviértase en distribuidor de nuestros productos.

////////////////////////////////////////////////////////////////

ENVÍOS

* Mercado Envíos
* Moto en el día: CABA envío GRATIS!; GBA consultar costos
* OCA a todo el país: GRATIS!
* Micro a terminal

////////////////////////////////////////////////////////////////

RETIROS EN NUESTRO SHOWROOM:

Sucursal Bidcom MICROCENTRO (Cerca de Lavalle y Florida). 
Horarios: Lunes a Viernes de 9 a 19hs / Sábados y Feriados 10 a 17hs

Estamos abiertos al público en general, no es necesario que nos avises antes de venir. Estamos abiertos de corrido en los horarios de atención. Vení a conocer nuestros mejores productos. 

////////////////////////////////////////////////////////////////

MEDIOS DE PAGO:

Podés abonar con todos los medios de pago habilitados con Mercado Pago:
 * Tarjeta de Crédito
 * Tarjeta de Débito
 * Pago Fácil
 * Rapi Pago
 * Provincia Net
 * Cobro Express
 * LINK
 * Carga Virtual

¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯
Emitimos Factura A y B.
En el caso de necesitar Factura A por favor comunicalo por el chat interno de Mercado Libre una vez realizada la compra.

////////////////////////////////////////////////////////////////

NUESTROS PRODUCTOS:

En el siguiente link podés ver todos los productos que desde Bidcom publicamos en MercadoLibre:

https://listado.mercadolibre.com.ar/gadnic_Tienda_gadnic_af_to

Tablets - Notebooks - Drones - Cámaras de Seguridad - Proyectores - Handies - Cámaras de deportes - TV Box - Contadoras de Billetes - Aspiradoras Robot - Smartwatches - Lentes VR - Patinetas Eléctricas - DJI - Órganos Eléctricos - Autos a Radio Control - Tocadiscos - y mucho más.

////////////////////////////////////////////////////////////////

ATENCIÓN: 
¡ÚLTIMAS UNIDADES A PRECIO DE SUPER OFERTA! 

Comprá ahora y asegurate tu unidad a este precio. ¡Aun Tenemos Stock!
¡Si podés ver esta publicación es porque TENEMOS STOCK, no hace falta que nos consultes!

¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯

Hacenos todas las consultas que tengas, para nosotros es un placer ayudarte.

Estamos Online para todo lo que necesites

BIDCOM

      </span>
    </div>
  </div>
  <div
    class="sc-bdVaJa jcenKy"
  >
    <div
      class="sc-bdVaJa fHkBQb"
    >
      <span>
        Nuevo - 
        250
         vendidos
      </span>
    </div>
    <div
      class="sc-bdVaJa ilQqel"
    >
      <span>
        Notebook Gadnic 15,6 Intel Windows 10 4gb Cloudbook  + Funda
      </span>
    </div>
    <div
      class="sc-bdVaJa sc-bxivhb jGPePi"
    >
      <span>
        $ 
        9.999
      </span>
      <small>
        00
      </small>
    </div>
    <button
      class="sc-bwzfXH dNrxNf"
      data-testid="button-buy"
    >
      Comprar
    </button>
  </div>
</div>
`;
